mlrun/runtimes/local.py

```diff
@@ -167,8 +177,12 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
             host=socket.gethostname(),
         )
         global_context.set(context)
+        # Running tracking services pre run to detect if some of them should be used:
+        trackers_manager = self._get_trackers_manager()
+        trackers_manager.pre_run(context)
         sout, serr = exec_from_params(handler, runobj, context, self.spec.workdir)
         log_std(self._db_conn, runobj, sout, serr, show=False)
```

Comment:
`log_std ` silent defaults to `False` here. That means, if local run execution fails, the `post_run` wont execute. is that acceptable?

Reply:
hmmm I guess we should move it so it will be called. (maybe we can log what is available). 

@ZeevRispler please move the line below here (pay attention to other calls in the code) and add a test that fails the run but log some stuff before and see if we catch it regardless 👍